azext_edge/edge/util/x509.py

```diff
@@ -73,3 +73,13 @@ def generate_self_signed_cert(valid_days: int = DEFAULT_VALID_DAYS) -> Tuple[byt
     )
 
     return (cert.public_bytes(serialization.Encoding.PEM), key_bytes)
+
+
+def decode_der_certificate(der_data) -> x509.Certificate:
+    # Decodes a DER-encoded X.509 certificate.
+    try:
+        cert = x509.load_der_x509_certificate(der_data, default_backend())
+        return cert
+    except Exception as e:
+        print(f"Error decoding DER certificate: {e}")
```

Comment:
We shouldn't use print here

Reply:
An exception is immediately raised from the consumer when this function doesn't have a result